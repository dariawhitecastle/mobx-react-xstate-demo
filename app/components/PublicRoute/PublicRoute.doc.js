import { describe, PropTypes } from 'react-desc'

export const doc = PublicRoute => {
  const DocumentedPublicRoute = describe(PublicRoute)
    .description('A PublicRoute')
    .usage(
      `import { PublicRoute } from 'components/PublicRoute';
<PublicRoute />`,
    )

  DocumentedPublicRoute.propTypes = {
    // this should be PropTypes.elementType but PropTypes imported from
    // react-desc do not support that type
    component: PropTypes.func.description(
      'Component type that should be rendered for the specified path',
    ),
  }

  return DocumentedPublicRoute
}
