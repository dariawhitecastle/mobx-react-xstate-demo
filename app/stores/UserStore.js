import { action, observable } from 'mobx'
import { ignore } from 'mobx-sync'
import { createContext } from 'react'

export class UserStore {
  @observable user = {}

  @observable tempUser = {}

  @observable token = ''

  @ignore
  @observable
  isLoading = false

  @ignore
  @observable
  message = ''

  @ignore
  @observable
  error = ''

  @observable message = ''

  @action.bound
  setCurrentUser({ token = '', ...user } = {}) {
    this.token = token
    this.user = {
      ...user,
    }
  }
}

export const user = new UserStore()
export const UserStoreContext = createContext(user)
