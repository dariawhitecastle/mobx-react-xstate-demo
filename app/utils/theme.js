import colors from './colors'
const fontFamily = 'Bryant Pro'

const theme = {
  global: {
    font: {
      family: `${fontFamily}, 'Helvetica Neue', Helvetica, Arial, sans-serif`,
      color: colors.primaryText /** @todo: this doesn't work */,
    },
    colors: {
      brand: colors.primary,
      focus: colors.primary50,
      'status-ok': colors.success,
      'status-error': colors.error,
    },
  },
  formField: {
    label: {
      size: 'small',
      weight: 600,
      color: '#2c3644',
      margin: { vertical: '0', bottom: '5px', horizontal: '0' },
    },
    background: '#ffffff',
    margin: { vertical: '0', bottom: '18px', horizontal: '0' },
  },
  radioButton: {
    extend: {
      minHeight: 24,
      minWidth: 24,
    },
  },
  table: {
    body: {
      pad: 'xsmall',
      font: {
        weight: 'normal',
        color: colors.primaryText,
      },
    },
    header: {
      align: 'center',
      fill: 'horizontal',
      pad: 'xsmall',
      verticalAlign: 'bottom',
      background: {
        color: colors.secondary10,
      },
    },
  },
  button: {
    border: {
      radius: '4px',
    },
    padding: {
      vertical: '13px',
    },
  },
  meter: {
    color: colors.primary10,
  },
}

export default theme
export { fontFamily }
