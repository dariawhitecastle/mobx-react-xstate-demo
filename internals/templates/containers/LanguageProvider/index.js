export { default as LanguageProvider } from './LanguageProvider'
