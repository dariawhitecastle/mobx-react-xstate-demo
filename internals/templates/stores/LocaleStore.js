import { observable, action } from 'mobx'
import { createContext } from 'react'

import { DEFAULT_LOCALE } from '../i18n'

export class LocaleStore {
  @observable locale = DEFAULT_LOCALE

  @action.bound
  changeLocale(locale) {
    this.locale = locale
  }
}

export const locale = new LocaleStore()
export const LocaleStoreContext = createContext(locale)
