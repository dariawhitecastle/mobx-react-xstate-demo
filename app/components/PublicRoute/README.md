<!-- ! This is a generated file. To make changes, edit <Component>.doc.js ! -->
## undefined
A PublicRoute

## Usage

```javascript
import { PublicRoute } from 'components/PublicRoute';
<PublicRoute />
```

## Properties

**component**

Component type that should be rendered for the specified path

```
function
```
  