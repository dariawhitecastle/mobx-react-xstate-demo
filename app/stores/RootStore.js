import { ignore } from 'mobx-sync'

import { locale } from './LocaleStore'

import { user } from './UserStore'

export class RootStore {
  @ignore storeLoaded = false

  locale = locale

  user = user
}

export const store = new RootStore()
