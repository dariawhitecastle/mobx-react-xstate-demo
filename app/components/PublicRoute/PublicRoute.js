/**
 * PublicRoute
 *
 * If a user is authenticated and attempts to access the component, they will be
 * redirected to their dashboard
 */

import React, { useContext } from 'react'
import { Route, Redirect } from 'react-router-dom'
import { observer } from 'mobx-react'

import { UserStoreContext } from '../../stores/UserStore'

const PublicRoute = observer(({ component: Component, ...rest }) => {
  const userStore = useContext(UserStoreContext)
  const { token } = userStore

  return (
    <Route
      {...rest}
      render={props =>
        token ? <Redirect to={{ pathname: '/dashboard' }} /> : <Component {...props} />
      }
    />
  )
})

export default PublicRoute
