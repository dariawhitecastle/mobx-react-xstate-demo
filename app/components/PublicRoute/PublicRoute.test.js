/**
 *
 * Tests for PublicRoute
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { IntlProvider } from 'react-intl'
import { MemoryRouter } from 'react-router-dom'
import { render, waitForElement } from '@testing-library/react'
import { Grommet } from 'grommet'
import 'jest-dom/extend-expect'

import { Register } from 'containers/Register'
import messages from 'containers/Register/Register.messages'
import theme from '../../utils/theme'
import { DEFAULT_LOCALE } from '../../i18n'
import PublicRoute from './PublicRoute'
import { UserStore, UserStoreContext } from '../../stores/UserStore'
import trunk from '../../configureStore'

describe('<PublicRoute />', () => {
  let userStore = null

  beforeAll(() => {
    userStore = new UserStore()
    trunk.init()
  })

  const renderComponent = store =>
    render(
      <Grommet theme={theme}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <MemoryRouter initialEntries={['/create-account']}>
            <UserStoreContext.Provider value={store}>
              <PublicRoute path="/create-account" component={Register} />
            </UserStoreContext.Provider>
          </MemoryRouter>
        </IntlProvider>
      </Grommet>,
    )

  it('renders Register if user is not authenticated', async () => {
    const { container, getByTestId } = renderComponent(userStore)
    expect(container).toMatchSnapshot()
    expect(container).toHaveTextContent(messages.header.defaultMessage)

    const registerPage = await waitForElement(() => getByTestId('register-page'))
    expect(registerPage).toHaveTextContent(messages.header.defaultMessage)
  })

  it('does not render Register if user is authenticated', async () => {
    userStore = {
      token: 'faketoken',
    }
    const { container } = renderComponent(userStore)
    expect(container).toMatchSnapshot()
    expect(container).not.toHaveTextContent(messages.header.defaultMessage)
  })
})
