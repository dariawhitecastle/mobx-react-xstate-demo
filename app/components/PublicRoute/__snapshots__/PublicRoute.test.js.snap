// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PublicRoute /> does not render Register if user is authenticated 1`] = `
<div>
  <div
    class="StyledGrommet-sc-19lkkz7-0 jjKgqr"
  />
</div>
`;

exports[`<PublicRoute /> renders Register if user is not authenticated 1`] = `
<div>
  <div
    class="StyledGrommet-sc-19lkkz7-0 jjKgqr"
  >
    <div
      class="StyledGrid-sc-1wofa1l-0 kwpRnp"
      data-testid="register-page"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 blgRIJ"
      >
        <h2
          class="StyledHeading-sc-1rdh4aw-0 cwhZKY"
        >
          <span>
            Create your account
          </span>
        </h2>
        <a
          class="StyledAnchor-sc-1rp7lwl-0 byullG"
          href="/create-account"
        >
          Start over
        </a>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 bKIioc"
      >
        <form>
          <div
            class="FormFieldstyles__StyledFormField-akn550-0 dEYomq FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 hEQCXC"
          >
            <label
              class="StyledText-sc-1sadyjn-0 gYQWKG"
            >
              Email address
            </label>
            <div
              class="StyledBox-sc-13pk1d4-0 gZSidW"
            >
              <div
                class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
              >
                <input
                  autocomplete="off"
                  class="StyledTextInput-sc-1x30a0s-0 duvNfq"
                  label="Email address"
                  name="email"
                  placeholder="Email"
                  type="email"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="FormFieldstyles__StyledFormField-akn550-0 dEYomq FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 hEQCXC"
          >
            <label
              class="StyledText-sc-1sadyjn-0 gYQWKG"
            >
              Password
            </label>
            <div
              class="StyledBox-sc-13pk1d4-0 gZSidW"
            >
              <div
                class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
              >
                <input
                  autocomplete="off"
                  class="StyledTextInput-sc-1x30a0s-0 duvNfq"
                  label="Password"
                  name="password"
                  placeholder="Password (Min 6 characters)"
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="FormFieldstyles__StyledFormField-akn550-0 dEYomq FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 hEQCXC"
          >
            <label
              class="StyledText-sc-1sadyjn-0 gYQWKG"
            >
              Confirm Password
            </label>
            <div
              class="StyledBox-sc-13pk1d4-0 gZSidW"
            >
              <div
                class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
              >
                <input
                  autocomplete="off"
                  class="StyledTextInput-sc-1x30a0s-0 duvNfq"
                  label="Confirm Password"
                  name="password_confirm"
                  placeholder="Type your password again"
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="StyledBox-sc-13pk1d4-0 iUevOx"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 cCLdcY"
            >
              <button
                class="StyledButton-sc-323bzc-0 MPVH"
                type="submit"
              >
                Back
              </button>
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 hyyRSo"
              />
              <button
                class="StyledButton-sc-323bzc-0 MPVH"
                style="max-width: 600px;"
                type="submit"
              >
                <span>
                  Create My Account & Basic Plan
                </span>
              </button>
            </div>
            <div
              class="FormFieldstyles__StyledFormField-akn550-0 dEYomq FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 hEQCXC"
              style="align-items: center;"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 gZSidW"
              >
                <label
                  class="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 bVmNDQ"
                >
                  <div
                    class="StyledCheckBox-sc-1dbk5ju-6 jXdLin StyledBox-sc-13pk1d4-0 emCRMK"
                  >
                    <input
                      class="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 kCsbbU"
                      name="terms"
                      type="checkbox"
                    />
                    <div
                      class="StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 fxKnpz StyledBox-sc-13pk1d4-0 jFlQDV"
                    />
                  </div>
                  <span>
                    I agree to the Terms & Conditions
                  </span>
                </label>
              </div>
            </div>
          </div>
        </form>
        <h5
          class="StyledHeading-sc-1rdh4aw-0 dvRIHC"
        />
      </div>
    </div>
  </div>
</div>
`;
